bot/exts/utils/reminders.py

```diff
@@ -450,35 +453,76 @@ async def edit_reminder(self, ctx: Context, id_: int, payload: dict) -> None:
         )
         await self._reschedule_reminder(reminder)
 
-    @remind_group.command("delete", aliases=("remove", "cancel"))
     @lock_arg(LOCK_NAMESPACE, "id_", raise_error=True)
-    async def delete_reminder(self, ctx: Context, id_: int) -> None:
-        """Delete one of your active reminders."""
-        if not await self._can_modify(ctx, id_):
-            return
+    async def _delete_reminder(self, ctx: Context, id_: int) -> bool:
+        """Acquires a lock on `id_` and returns `True` if reminder is deleted, otherwise `False`."""
+        if not await self._can_modify(ctx, id_, send_on_denial=False):
+            return False
 
         await self.bot.api_client.delete(f"bot/reminders/{id_}")
         self.scheduler.cancel(id_)
+        return True
 
-        await self._send_confirmation(
-            ctx,
-            on_success="That reminder has been deleted successfully!",
-            reminder_id=id_
+    @remind_group.command("delete", aliases=("remove", "cancel"))
+    async def delete_reminder(self, ctx: Context, ids: Greedy[int]) -> None:
+        """Delete one of your active reminders."""
+        deleted_ids = []
+        for id_ in ids:
+            try:
```

Comment:
Could consider making `ids` a set to remove duplicates.

I also wonder if we'd want an upper limit on the length to prevent spamming the site api (you could currently cause nearly 1000 api requests with `!remind delete 1 1 1 ...`)

Reply:
Yep, agreed :+1:

I've set the limit to 5, since that's high enough to likely not inconvenience the user, but l low enough to still prevent API abuse.